tests/gtest/avifstreamtest.cc

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <algorithm>
+#include <cstdint>
+#include <vector>
+
+#include "avif/internal.h"
+#include "aviftest_helpers.h"
+#include "gtest/gtest.h"
+
+namespace libavif {
+namespace {
+
+//------------------------------------------------------------------------------
+
+TEST(StreamTest, Roundtrip) {
+  // TODO(yguyon): Check return values once pull request #1498 is merged.
+
+  // Write some fields.
+  testutil::AvifRwData rw_data;
+  avifRWStream rw_stream;
+  avifRWStreamStart(&rw_stream, &rw_data);
+  EXPECT_EQ(avifRWStreamOffset(&rw_stream), size_t{0});
+
+  const uint8_t rw_somedata[] = {3, 1, 4};
+  avifRWStreamWrite(&rw_stream, rw_somedata, sizeof(rw_somedata));
+
+  const char rw_somechars[] = "somechars";
+  avifRWStreamWriteChars(&rw_stream, rw_somechars, sizeof(rw_somechars));
+
+  const char rw_box_type[] = "type";
+  avifBoxMarker rw_box_marker =
+      avifRWStreamWriteBox(&rw_stream, rw_box_type, /*contentSize=*/0);
+
+  const uint8_t rw_someu8 = 0xA;
```

Comment:
A hexadecimal digit is 4 bits, so `uint8_t` may have up to 2 nonzero hexadecimal digits. Similarly for the initializers for `rw_someu16`, etc.